### PR TITLE
dxDataGrid: Fix accessibility table structure (T595044)

### PR DESCRIPTION
**Pull request by AntonSermyazhko**

### CODE REVIEW

File: js/ui/grid_core/ui.grid_core.columns_view.js [hunk 1]
```diff
@@ -450,16 +450,18 @@ exports.ColumnsView = modules.View.inherit(columnStateMixin).inherit({
...
 
         options.row.cells.push(cellOptions);
 
         $cell = that._createCell(cellOptions);
 
-        // TODO move to cellPrepared
-        if(!column.command) {
-            that.setAria("label", that.localize("dxDataGrid-ariaColumn") + " " + column.caption + ", " + that.localize("dxDataGrid-ariaValue") + " " + cellOptions.text, $cell);
+        if(cellOptions.rowType !== "freeSpace") {
+            that.component.setAria({
```
Comment (vconst) [inline js/ui/grid_core/ui.grid_core.columns_view.js]: Do not use that.component.setArea. The that.setAria is faster.

File: js/ui/grid_core/ui.grid_core.columns_view.js [hunk 2]
```diff
@@ -152,29 +151,29 @@ exports.ColumnsView = modules.View.inherit(columnStateMixin).inherit({
...
-        return $("<tr>")
-            .addClass(ROW_CLASS)
-            .attr("role", "row");
-    },
-
-    _getTableRoleName: function() {
-        return "grid";
+    _createRow: function(rowObject) {
+        var $element = $("<tr>").addClass(ROW_CLASS);
+        if(rowObject) {
```
Comment (vconst) [inline js/ui/grid_core/ui.grid_core.columns_view.js]: Why is this change needed?

File: js/ui/grid_core/ui.grid_core.grid_view.js [hunk 3]
```diff
@@ -618,17 +618,24 @@ var GridView = modules.View.inherit({
...
 
         $groupElement.addClass(GRIDBASE_CONTAINER_CLASS);
         $groupElement.toggleClass(that.addWidgetPrefix(BORDERS_CLASS), !!that.option("showBorders"));
+
+        that.component.setAria({ "role": "presentation" }, $rootElement);
+
         that.component.setAria({
-            "role": "application",
+            "role": this._getTableRoleName(),
             "label": messageLocalization.format(that._getWidgetAriaLabel())
```
Comment (vconst) [inline js/ui/grid_core/ui.grid_core.grid_view.js:646]: Is this label needs if we add good role?

File: js/ui/grid_core/ui.grid_core.rows.js [hunk 4]
```diff
@@ -664,7 +682,7 @@ module.exports = {
                         .toggleClass(COLUMN_LINES_CLASS, that.option("showColumnLines"));
 
                     for(i = 0; i < columns.length; i++) {
-                        freeSpaceRowElement.append(that._createCell({ column: columns[i], rowType: "freeSpace", columnIndex: i, columns: columns }));
+                        freeSpaceRowElement.append(that._createCell({ column: columns[i], rowType: "freeSpace" }));
```
Comment (vconst) [inline js/ui/grid_core/ui.grid_core.rows.js]: Why? This change may course problems with the new advancedRendering.

File: testing/tests/DevExpress.ui.widgets.dataGrid/dataGrid.tests.js [hunk 5]
```diff
@@ -2468,6 +2484,99 @@ QUnit.test("height 100% when parent container with fixed height when virtual scr
...
+    }).dxDataGrid("instance");
+
+    clock.tick();
+
+    rowsView = dataGrid._views.rowsView;
+    rows = rowsView.element().find(".dx-row").filter(function(index, element) { return !$(element).hasClass("dx-freespace-row"); });
+
+    // assert
+    rowIndexOffset = dataGrid._controllers.data.getRowIndexOffset();
+    for(i = 0; i < rows.length; ++i) {
```
Comment (vconst) [inline testing/tests/DevExpress.ui.widgets.dataGrid/dataGrid.tests.js]: It is not necessary to check all rows. Also, I does not see reasons to use private API getRowIndexOffset in this test.

File: testing/tests/DevExpress.ui.widgets.dataGrid/rowsView.tests.js [hunk 6]
```diff
@@ -143,30 +143,6 @@ QUnit.test('Create col elements by columns collection', function(assert) {
     assert.equal(cols[4].style.width, '91px', '5 column width');
 });
 
-QUnit.test('Check WAI-ARIA cell labels', function(assert) {
```
Comment (vconst) [inline testing/tests/DevExpress.ui.widgets.dataGrid/rowsView.tests.js:146]: It is better not to remove the test, but to change it to the area-colindex and selected attributes, There are no other tests for these.

**Review by vconst — CHANGES_REQUESTED** [review]

File: js/ui/grid_core/ui.grid_core.columns_view.js (hunk 1 above)
Comment (AntonSermyazhko) [reply js/ui/grid_core/ui.grid_core.columns_view.js]: Ok, than I need to invoke 3 times "that.setAria"

File: js/ui/grid_core/ui.grid_core.columns_view.js (hunk 2 above)
Comment (AntonSermyazhko) [reply js/ui/grid_core/ui.grid_core.columns_view.js]: Because I moved setting role="grid" to the "grid.core" for the base container and function "_getTableRoleName" moved too.

File: js/ui/grid_core/ui.grid_core.grid_view.js (hunk 3 above)
Comment (AntonSermyazhko) [reply js/ui/grid_core/ui.grid_core.grid_view.js:646]: I suppose this can be useful to understand what grid was selected..

File: js/ui/grid_core/ui.grid_core.rows.js (hunk 4 above)
Comment (AntonSermyazhko) [reply js/ui/grid_core/ui.grid_core.rows.js]: My fault, will roll back it

File: testing/tests/DevExpress.ui.widgets.dataGrid/dataGrid.tests.js (hunk 5 above)
Comment (AntonSermyazhko) [reply testing/tests/DevExpress.ui.widgets.dataGrid/dataGrid.tests.js]: Ok. I will change it to check constant values

File: testing/tests/DevExpress.ui.widgets.dataGrid/rowsView.tests.js (hunk 6 above)
Comment (AntonSermyazhko) [reply testing/tests/DevExpress.ui.widgets.dataGrid/rowsView.tests.js:146]: Such kind of tests already exists: rowsView.tests.js (Check WAI-ARIA attributes for data rows/cells after render rows)

**Review by vconst — DISMISSED** [review]

**Review by vconst — APPROVED** [review]
